src/main/java/coresearch/cvurl/io/request/ProxyAwareCVurlRequest.java

```diff
@@ -0,0 +1,132 @@
+        return removeProxiesAfterAction(request.asyncAsObject(type, statusCode));
+    }
+
+    @Override
+    public <T> CompletableFuture<T> asyncAsObject(BodyType<T> type, int statusCode) {
+        return removeProxiesAfterAction(request.asyncAsObject(type, statusCode));
+    }
+
+    @Override
+    public <T> CompletableFuture<T> asyncAsObject(Class<T> type) {
+        return removeProxiesAfterAction(request.asyncAsObject(type));
+    }
+
+    @Override
+    public <T> CompletableFuture<T> asyncAsObject(BodyType<T> type) {
+        return removeProxiesAfterAction(request.asyncAsObject(type));
+    }
+
+    @Override
+    public CompletableFuture<Response<String>> asyncAsString() {
+        return removeProxiesAfterAction(request.asyncAsString());
+    }
+
+    @Override
+    public CompletableFuture<Response<String>> asyncAsString(HttpResponse.PushPromiseHandler<String> pph) {
+        return removeProxiesAfterAction(request.asyncAsString(pph));
+    }
+
+    @Override
+    public CompletableFuture<Response<InputStream>> asyncAsStream() {
+        return removeProxiesAfterAction(request.asyncAsStream());
+    }
+
+    @Override
+    public CompletableFuture<Response<InputStream>> asyncAsStream(HttpResponse.PushPromiseHandler<InputStream> pph) {
+        return removeProxiesAfterAction(request.asyncAsStream(pph));
+    }
+
+    @Override
+    public <T> CompletableFuture<Response<T>> asyncAs(HttpResponse.BodyHandler<T> bodyHandler) {
+        return removeProxiesAfterAction(request.asyncAs(bodyHandler));
+    }
+
+    @Override
+    public <T> CompletableFuture<Response<T>> asyncAs(HttpResponse.BodyHandler<T> bodyHandler, HttpResponse.PushPromiseHandler<T> pph) {
+        return removeProxiesAfterAction(request.asyncAs(bodyHandler, pph));
+    }
+
+    @Override
+    public <T> Optional<T> asObject(Class<T> type, int statusCode) {
+        return removeProxiesAfterAction(() -> request.asObject(type, statusCode));
+    }
+
+    @Override
+    public <T> Optional<T> asObject(BodyType<T> type, int statusCode) {
+        return removeProxiesAfterAction(() -> request.asObject(type, statusCode));
+    }
+
+    @Override
+    public <T> T asObject(Class<T> type) {
+        return removeProxiesAfterAction(() -> request.asObject(type));
+    }
+
+    @Override
+    public <T> T asObject(BodyType<T> type) {
+        return removeProxiesAfterAction(() -> request.asObject(type));
+    }
+
+    @Override
+    public Optional<Response<String>> asString() {
+        return removeProxiesAfterAction(request::asString);
+    }
+
+    @Override
+    public Optional<Response<InputStream>> asStream() {
+        return removeProxiesAfterAction(request::asStream);
+    }
+
+    @Override
+    public <T> Optional<Response<T>> as(HttpResponse.BodyHandler<T> bodyHandler) {
+        return removeProxiesAfterAction(() -> request.as(bodyHandler));
+    }
+
+    private <T> T removeProxiesAfterAction(Supplier<T> action) {
```

Comment:
We need to find a better way for handling proxy per request. I think it could be handled by our custom ProxySelector. For example: after the execution of the select method proxy per request could be cleaned up

Reply:
Fixed